pypeit/flatfield.py

```diff
@@ -380,9 +329,573 @@ def load(self, ifile=None):
+#                ax.set_ylim(-0.05, 0.05)
+                ax.legend()
+                ax.set_xlabel('Spectral Pixel')
+                ax.set_ylabel('Residuals from pixelflat 2-d fit')
+                ax.set_title('Spectral Residuals for slit={:d}'.format(slit))
+                plt.show()
+
+                plt.clf()
+                ax = plt.gca()
+                ax.plot(twod_spat_coo_data[goodpix], resid[goodpix], color='k', marker='o',
+                        markersize=0.2, mfc='k', fillstyle='full', linestyle='None',
+                        label='good points')
+                ax.plot(twod_spat_coo_data[badpix], resid[badpix], color='red', marker='+',
+                        markersize=0.5, mfc='red', fillstyle='full', linestyle='None',
+                        label='masked')
+                ax.axhline(twod_sigrej*twod_sig, color='lawngreen', linestyle='--',
+                           label='rejection thresholds', zorder=10, linewidth=2.0)
+                ax.axhline(-twod_sigrej*twod_sig, color='lawngreen', linestyle='--', zorder=10,
+                           linewidth=2.0)
+#                ax.set_ylim((-0.05, 0.05))
+#                ax.set_xlim(-0.02, 1.02)
+                ax.legend()
+                ax.set_xlabel('Normalized Slit Position')
+                ax.set_ylabel('Residuals from pixelflat 2-d fit')
+                ax.set_title('Spatial Residuals for slit={:d}'.format(slit))
+                plt.show()
+
+            # Save the 2D residual model
+            twod_model = np.ones_like(rawflat)
+            if exit_status > 1:
+                msgs.warn('Two-dimensional fit to flat-field data failed!  No higher order '
+                          'flat-field corrections included in model of slit {0}!'.format(slit))
+            else:
+                twod_model[twod_gpm] = twod_flat_fit[np.argsort(twod_srt)]
+
+            # Construct the full flat-field model
+            # TODO: Why is the 0.05 here for the illumflat compared to the 0.01 above?
+            self.flat_model[onslit] = twod_model[onslit] \
+                                        * np.fmax(self.msillumflat[onslit], 0.05) \
+                                        * np.fmax(spec_model[onslit], 1.0)
+
+            # Construct the pixel flat
+            self.mspixelflat[onslit] = rawflat[onslit]/self.flat_model[onslit]
+            # TODO: Add some code here to treat the edges and places where fits
+            # go bad?
+
+        # Update the tilts dictionary if the slit edges were tweaked
+        if tweak_slits:
+            self.tilts_dict['tilts'] = tweaked_tilts
+
+        # Set the pixelflat to 1.0 wherever the flat was nonlinear
+        self.mspixelflat[rawflat >= nonlinear_counts] = 1.0
+        # Do not apply pixelflat field corrections that are greater than
+        # 100% to avoid creating edge effects, etc.
+        # TODO: Should we do the same for the illumflat?
+        self.mspixelflat = np.clip(self.mspixelflat, 0.5, 2.0)
+
+
```

Comment:
You need to add the analog of the following lines of code here that were in the previous flatfield.py on lines 278-280

            self.mspixelflat[thismask_out] = pixelflat[thismask_out]
            self.msillumflat[thismask_out] = illumflat[thismask_out]
            self.flat_model[thismask_out] = flat_model[thismask_out]

In other words, all of the flat field images should be assigned to only the pixels that lie on the new tweaked slits. That is what thismask_out encoded before. For example your illumination flat is defined everywhere within the padded (not-tweaked) slit boundaries. This causes edge artifacts like these: 

![Screen Shot 2020-03-15 at 1 03 30 PM](https://user-images.githubusercontent.com/8774563/76709994-c9959980-66c0-11ea-90c6-4c9be214fb2a.png)

Which causes problems with CR rejection.

Reply:
Fixed in the next push.  Thanks for this inspection!
<img width="1041" alt="fixed_illum" src="https://user-images.githubusercontent.com/15785898/76897407-efea3f00-6850-11ea-8b8d-a79039635236.png">